### PR TITLE
Set associations

### PR DESCRIPTION
**Pull request by Ukaypromise**
### OOP school library: Set up Associations

- [ ] For this pull request I have implemented the following requirements:

- Create a class Classroom, Book, Rental.
- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
- Create the has-many side of Book and Rental (a book has many rentals).
- Create the belongs-to side of Rental and Book (a rental belongs to a book).
- Create the has-many side of Person and Rental (a person has many rentals).
- Create the belongs-to side of Rental and Person (a rental belongs to a person).
- Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Classroom
+  attr_accessor :label
+  attr_reader :students
```
Comment (mateo951) [inline classroom.rb:3]: has-many / belongs-to relationships between Clasroom and Student is created 😎

File: student.rb [hunk 2]
```diff
@@ -9,4 +9,9 @@ def initialize(age, classroom, name = 'Unknown', parent_permission: true)
   def play_hooky
     '¯\(ツ)/¯'
   end
+
+  def classroom(classes)
+    @classroom = classes
+    classes.students.push(self) unless classes.students.include(self)
```
Comment (mateo951) [inline student.rb:15]: When setting the classroom for a student it also adds it to the classrooms students ✔

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author, :rentals
```
Comment (mateo951) [inline book.rb:2]: `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental` 🔥

**Review by mateo951 — APPROVED** [review]
### Hi @Ukaypromise, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🐐 
### Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ukaypromise** [comment]
This is a beautiful review @mateo951

**Pull request merged**